pkg/provider/manager/manager.go

```diff
@@ -129,17 +133,32 @@ func (m *ProviderManager) GetProviders() map[string]Provider {
 	return providers
 }
 
-func (m *ProviderManager) RegisterProvider(pluginPath string) error {
+func (m *ProviderManager) RegisterProvider(pluginPath string, manualInstall bool) error {
 	pluginRef, err := m.initializeProvider(pluginPath)
 	if err != nil {
 		return err
 	}
 
 	m.pluginRefs[pluginRef.name] = pluginRef
 
-	p, err := m.dispenseProvider(pluginRef.client, pluginRef.name)
+	lockFilePath := filepath.Join(pluginRef.path, INITIAL_SETUP_LOCK_FILE_NAME)
+	_, err = os.Stat(lockFilePath)
+	if os.IsNotExist(err) || manualInstall {
+		err = m.SetPresetTargets(pluginRef.name)
+		if err != nil {
+			return err
+		}
+	}
+
+	log.Infof("Provider %s initialized", pluginRef.name)
+
+	return nil
+}
+
+func (m *ProviderManager) SetPresetTargets(providerName string) error {
```

Comment:
Seems to me that this can be a private function (even though it might not even have to be a function since it seems to be only called in one place.

Reply:
Okay, I removed the function and it looks fine